Resources/public/js/views/services/plugins/ez-locationcreateplugin.js

```diff
@@ -0,0 +1,156 @@
+        initializer: function () {
+            this.onHostEvent('*:createLocation', this._createLocationSelect);
+        },
+
+        /**
+         * createLocation event handler, launch the universal discovery widget
+         * to choose a parent location(s) for new location(s) of given content
+         *
+         * @method _createLocationSelect
+         * @private
+         * @param {EventFacade} e createLocation event facade
+         */
+        _createLocationSelect: function (e) {
+            var service = this.get('host');
+
+            service.fire('contentDiscover', {
+                config: {
+                    title: "Select the location where you want to create new location",
+                    contentDiscoveredHandler: Y.bind(this._createLocation, this),
+                    multiple: true,
+                    data: {
+                        afterCreateCallback: e.afterCreateCallback
+                    }
+                },
+            });
+        },
+
+        /**
+         * Creates new location as a descendant of selected location
+         *
+         * @method _createLocation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _createLocation: function (e) {
+            var service = this.get('host'),
+                capi = service.get('capi'),
+                contentService = capi.getContentService(),
+                content = service.get('content'),
+                locationsCreatedCounter = 0,
+                notificationIdentifier = 'create-location-' + content.get('id'),
+                data = e.target.get('data'),
+                stack = new Y.Parallel(),
+                that = this;
+
+            this._notify(
+                "Creating new location for '" + content.get('name') + "'",
+                notificationIdentifier,
+                'started',
+                5
+            );
+
+            Y.Array.each(e.selection, function (selection) {
+                var parentLocationId = selection.location.get('id'),
+                    parentContent = selection.content,
+                    locationCreateStruct = contentService.newLocationCreateStruct(parentLocationId),
+                    errNotificationIdentifier = 'create-location-' + content.get('id') + '-' + parentContent.get('id'),
+                    end = stack.add( function (error) {
+                        if (error) {
+                            that._notify(
+                                "Creating new location for '" + content.get('name') + "' under '" + parentContent.get('name') + "' failed",
+                                errNotificationIdentifier,
+                                'error',
+                                0
+                            );
+                            return;
+                        }
+
+                        locationsCreatedCounter++;
+                    });
+
+                contentService.createLocation(content.get('id'), locationCreateStruct, end);
```

Comment:
I would rather create a `addLocation` method in the content model. Something like:
```
content.addLocation(options, parentLocation, callback);
```